platform/standalone.go

```diff
@@ -34,39 +34,58 @@ func (s *StandaloneMessageBus) Connect(inst flux.InstanceID) (Platform, error) {
 }
 
 // Subscribe introduces a Platform to the message bus, so that
-// requests can be routed to it. This will block until the connection
-// is closed, or replaced by another.
-func (s *StandaloneMessageBus) Subscribe(inst flux.InstanceID, p Platform) error {
+// requests can be routed to it. Once the connection is closed --
+// trying to use it is the only way to tell if it's closed -- the
+// error representing the cause will be sent to the channel supplied.
+func (s *StandaloneMessageBus) Subscribe(inst flux.InstanceID, p RemotePlatform, complete chan<- error) {
 	s.Lock()
 	// We're replacing another client
 	if existing, ok := s.connected[inst]; ok {
 		delete(s.connected, inst)
 		existing.closeWithError(errors.New("duplicate connection; replacing with newer"))
 	}
 
-	// Add our new client in
 	done := make(chan error)
 	s.connected[inst] = &removeablePlatform{
-		Platform: p,
-		done:     done,
+		remote: p,
+		done:   done,
 	}
 	s.Unlock()
 
-	// Wait to be kicked, or an error to happen
-	err := <-done
+	// The only way we detect remote platforms closing are if an RPC
+	// is attempted and it fails. When that happens, clean up behind
+	// us.
+	go func() {
```

Comment:
Or do you mean, why not just go back to how it was, where it blocked inside the method? Because then it's very difficult to write a test that uses Subscribe -- that was the motivation for this change.